src/main/java/org/sagebionetworks/bridge/spring/controllers/AuthenticationController.java

```diff
@@ -291,6 +292,26 @@ public StatusMessage resetPassword() throws Exception {
         authenticationService.resetPassword(passwordReset);
         return new StatusMessage("Password has been changed.");
     }
+    
+    @PostMapping("/v3/auth/oauth/signIn")
+    public JsonNode oauthSignIn() {
+        OAuthAuthorizationToken token = parseJson(OAuthAuthorizationToken.class);
+        
+        Study study = studyService.getStudy(token.getStudyId());
+        CriteriaContext context = getCriteriaContext(study.getStudyIdentifier());
+        
+        UserSession session = null;
+        try {
+            session = authenticationService.oauthSignIn(context, token);
+        } catch(ConsentRequiredException e) {
+            setCookieAndRecordMetrics(e.getUserSession());
+            throw e;
+        }
+        setCookieAndRecordMetrics(session);
+        verifySupportedVersionOrThrowException(study);
```

Comment:
Similarly, I don't think it's necessary to include the shut-off switch, since we aren't intending Synapse sign-in for participants.

Reply:
Removing. I think setCookieAndRecordMetrics should be moved into the request interceptor so that even when we throw RuntimeExceptions, we log the request correctly. We shouldn't have to catch like this all over the place. I will file a JIRA.

Reply:
https://sagebionetworks.jira.com/browse/BRIDGE-2632